ext/scheduler/airflow2/airflow.go

```diff
@@ -222,3 +206,91 @@ func (a *scheduler) Clear(ctx context.Context, projSpec models.ProjectSpec, jobN
 
 	return nil
 }
+
+func (a *scheduler) GetDagRunStatus(ctx context.Context, projSpec models.ProjectSpec, jobName string, startDate time.Time,
+	endDate time.Time, batchSize int) ([]models.JobStatus, error) {
+	schdHost, ok := projSpec.Config[models.ProjectSchedulerHost]
+	if !ok {
+		return nil, errors.Errorf("scheduler host not set for %s", projSpec.Name)
+	}
+	authToken, ok := projSpec.Secret.GetByName(models.ProjectSchedulerAuth)
+	if !ok {
+		return []models.JobStatus{}, errors.Errorf("%s secret not configured for project %s", models.ProjectSchedulerAuth, projSpec.Name)
+	}
+	schdHost = strings.Trim(schdHost, "/")
+	postURL := fmt.Sprintf("%s/%s", schdHost, dagStatusBatchUrl)
+
+	pageOffset := 0
+	var jobStatus []models.JobStatus
+	var responseJson struct {
+		DagRuns      []map[string]interface{} `json:"dag_runs"`
+		TotalEntries int                      `json:"total_entries"`
+	}
+
+	for {
+		dagRunBatchReq := fmt.Sprintf(`{
+		"page_offset": %d, 
+		"page_limit": %d, 
+		"dag_ids": ["%s"],
+		"execution_date_gte": "%s",
+		"execution_date_lte": "%s"
+		}`, pageOffset, batchSize, jobName, startDate.UTC().Format(airflowDateFormat), endDate.UTC().Format(airflowDateFormat))
+		var jsonStr = []byte(dagRunBatchReq)
+		request, err := http.NewRequest(http.MethodPost, postURL, bytes.NewBuffer(jsonStr))
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to build http request for %s", dagStatusBatchUrl)
+		}
+		request.Header.Set("Content-Type", "application/json")
+		request.Header.Set("Authorization", fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(authToken))))
+
+		resp, err := a.httpClient.Do(request)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to fetch airflow dag runs from %s", dagStatusBatchUrl)
+		}
+		if resp.StatusCode != http.StatusOK {
+			return nil, errors.Errorf("failed to fetch airflow dag runs from %s", dagStatusBatchUrl)
+		}
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to read airflow response")
+		}
+
+		err = json.Unmarshal(body, &responseJson)
+		if err != nil {
```

Comment:
when the return values are not used in further statements, its cleaner to merge them with `if err := ...; err != nil {}`